public/js/views/card-details.jsx

```diff
@@ -11,58 +9,25 @@ import { default as tracking } from 'tracking';
 export default class CardDetailsView extends Component {
 
   static propTypes = {
+    braintreeToken: PropTypes.string.isRequired,
     productId: PropTypes.string.isRequired,
   }
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      braintree_token: false,
-    };
-  }
-
   componentDidMount() {
-    console.log('Requesting braintree token');
-    // TODO: move this to a purchase action.
-
-    this.mounted = true;
-
     tracking.setPage('/card-details');
-
-    $.ajax({
-      method: 'post',
-      url: '/api/braintree/token/generate/',
-      context: this,
-    }).then(function(data) {
-      if (this.mounted) {
-        this.setState({'braintree_token': data.token}); // eslint-disable-line
-      }
-    }).fail(function() {
-      // TODO: some error state.
-      console.log('failed to get braintree token');
-    });
-  }
-
-  componentWillUnmount() {
-    this.mounted = false;
   }
 
   render() {
-
-    if (this.state.braintree_token) {
-      return (
-        <div className="card-details">
-          <ProductDetail productId={this.props.productId} />
-          <CardForm
-            data-token={this.state.braintree_token}
-            id="braintree-form"
-            method="post"
-            productId={this.props.productId}
-          />
-        </div>
-      );
-    } else {
-      return <Spinner />;
-    }
+    return (
+      <div className="card-details">
+        <ProductDetail productId={this.props.productId} />
+        <CardForm
+          data-token={this.props.braintreeToken}
```

Comment:
I think `data-token` is pre-react cruft. While you're in here, you might as well rename it to `braintreeToken={...}` because the CardForm code will read a lot cleaner.